python/runtime/model/model.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module saves or loads the SQLFlow model.
+"""
+import os
+from enum import Enum
+
+# TODO(yancey1989): move diag to the upper direction
+from runtime.tensorflow.diag import SQLFlowDiagnostic
+
+
+class EstimatorType(Enum):
+    """The enum type for various SQLFlow estimator.
+    """
+    TENSORFLOW = 1
+    XGBOOST = 2
+
+
+class Model(object):
+    """Model module represents a SQLFlow trained model, which includes
+three parts:
+1. the estimator type indicates which SQLFlow estimator comes from.
+2. the model meta indicates the meta data of training .e.g attributions,
+   feature column types.
+3. the model data indicated the trained model, which generated by the AI
+   engine, .e.g TensorFlow, XGBoost.
+
+Usage:
+
+    meta = runtime.collect_model_metadata(train_params={...},
+                                          model_params={...})
+    m = runtime.model.Model(ModelType.XGBOOST, meta)
+    m.save(uri="sqlfs://mydatabase/mytable")
+
+    """
+    def __init__(self, typ, meta):
+        """
+        Args:
+
+        typ, ModelType
+            the enum value of ModelType.
+
+        meta, JSON
+            the training meta with JSON format.
+        """
+        self._typ = typ
+        self._cwd = os.getcwd()
+        self._meta = meta
+
+    def save(self, uri, datasource=None):
+        """ save this model object.
+        Args:
+
+        uri, string
+            model saving URL with format: <driver>://<path>
+
+        datasource, string
+            the connection datasource DSN is required if saving with sqlfs.
+        """
+        driver, path = parseURI(uri)
+        if driver == "file":
+            # TODO(yancey1989): save model with local file system.
+            pass
+        elif driver == "sqlfs":
+            # TODO(yancey1989): save model with SQL file system.
+            pass
+        else:
+            raise SQLFlowDiagnostic("unsupported driven to save model: %s" %
+                                    driver)
+
+
+def parseURI(uri):
+    """Parse the model URI into two parts: driver and path.
+    """
+    array = uri.split("://")
```

Comment:
What does sqlfs schema look like?  sqlfs://mysql://xxx or mysql://xxx, it seem not compatible with `save` function.

Reply:
It's like `sqlfs://user/yanxu`, `mysq://xxx` is the datasource DNS format, and we can pass it to `model.save(uri="sqlfs://xxx", datasource="mysql://")`.